pyam/core.py

```diff
@@ -1361,11 +1412,27 @@ def _meta_idx(data):
     return data[META_IDX].drop_duplicates().set_index(META_IDX).index
 
 
-def _aggregate(df, by):
+def _aggregate(df, by, method='sum', weightvar_df=None):
     """Aggregate `df` by specified column(s), return indexed `pd.Series`"""
     by = [by] if isstr(by) else by
     cols = [c for c in list(df.columns) if c not in ['value'] + by]
-    return df.groupby(cols).sum()['value']
+    # pick aggregator func (default: sum)
+    _agg_func = _get_method_func(method)
+    return df.groupby(cols)['value'].agg(_agg_func)
+
+
+def _get_method_func(method):
+    if method == 'min':
```

Comment:
Would it make sense to check type of input parameter and use something like `getattr(np, method)` to collapse multiple if branches?